procrastinate/sql/migrations/delta_1.2.0_drop_version.sql

```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS procrastinate_version;
```

Comment:
Isn't that a breaking change according to our "semver-like" plan ? I think we can do 2.0.0.

(version numbers are cheap :) )

Reply:
You mean it's a breaking change because migrating the database would require updating the Procrastinate code?

Reply:
I guess so, but maybe this needs a better definition ? What should be considered breaking ?

Reply:
That's a good question. Here's my take on it:

I think what's important is the perspective of Procrastinate users. And I think a Procrastinate user would never attempt to migrate the Procrastinate database schema independently from upgrading Procrastinate (the Python package). What users want is upgrade to a more recent version of Procrastinate, which may imply migrating the Procrastinate database schema.

As I see it, a PATCH release of Procrastinate must not imply a data schema migration. A MINOR release of Procrastinate may imply a database schema migration, but the migration shouldn't be mandatory, meaning that Procrastinate would still work as expected if the user didn't migrate the database schema. And a MAJOR release of Procrastinate does imply a database migration, with the risk that Procrastinate won't function properly if the user didn't not apply the migration.

With this definition, the removal of a table (as done in this PR) is not a breaking change.

Reply:
Are you're talking about the procrastinate version or the schema version ?

Reply:
I am talking about the Procrastinate version. I think the Procrastinate version is what users see and care about the most.